### PR TITLE
Upgrade controllers Javascript syntax with esnext + manual tweaks

### PR DESCRIPTION
**Pull request by sedubois**

### CODE REVIEW

File: server/controllers/middlewares.js [hunk 1]
```diff
@@ -77,20 +79,20 @@ module.exports = function(app) {
...
 
-      var name, email, paypalEmail;
+      let name, email, paypalEmail;
 
       // TODO remove #postmigration, replaced by req.body.expense
       if (req.body.transaction) {
-        email = req.body.transaction.email;
-        paypalEmail = req.body.transaction.paypalEmail;
-        name = req.body.transaction.name;
+        ({ email } = req.body.transaction);
```
Comment (asood123) [inline server/controllers/middlewares.js]: didn't know you could do that. It treats all local vars as part of a single object? Are the parens on the outside needed?

**Comment by asood123** [comment]
Also, is there a way to enforce this syntax? We should add it to the eslint rules.

File: server/controllers/middlewares.js (hunk 1 above)
Comment (sedubois) [inline server/controllers/middlewares.js]: Didn't know either ;) and yes, the parentheses are expected, otherwise syntax error

**Comment by sedubois** [comment]
I'll have a look for eslint rules

**Pull request merged**